sdmetrics/multi_table/statistical/cardinality_statistic_similarity.py

```diff
@@ -0,0 +1,165 @@
+                cardinality distributions of that foreign key.
+        """
+        if set(real_data.keys()) != set(synthetic_data.keys()):
+            raise ValueError('`real_data` and `synthetic_data` must have the same tables.')
+        if metadata is None:
+            raise ValueError('`metadata` cannot be ``None``.')
+        if not isinstance(metadata, dict):
+            metadata = metadata.to_dict()
+
+        score_breakdowns = {}
+        for table_name, table in metadata['tables'].items():
+            for field_name, field in table['fields'].items():
+                if 'ref' in field:
+                    parent_table_name = field['ref']['table']
+                    parent_field_name = field['ref']['field']
+                    cardinality_real = get_cardinality_distribution(
+                        real_data[parent_table_name][parent_field_name],
+                        real_data[table_name][field_name],
+                    )
+                    cardinality_synthetic = get_cardinality_distribution(
+                        synthetic_data[parent_table_name][parent_field_name],
+                        synthetic_data[table_name][field_name],
+                    )
+                    score_breakdown = cls._compute_statistic(
+                        cardinality_real, cardinality_synthetic, statistic)
+                    score_breakdowns[(parent_table_name, table_name)] = score_breakdown
+
+        if len(score_breakdowns) == 0:
+            return {'score': np.nan}
+
+        return score_breakdowns
+
+    @classmethod
+    def compute(cls, real_data, synthetic_data, metadata=None, statistic='mean'):
+        """Compute the average of cardinality statistic similarity in the given tables.
+
+        Compute the average statistic similarity in cardinality distributions for
+        (parent, child) relationship in the given tables, as specified in the metadata.
+        The statistic similarity is computed based on the requested statistic.
+
+        Args:
+            real_data (dict[str, pandas.DataFrame]):
+                The tables from the real dataset, passed as a dictionary of
+                table names and pandas.DataFrames.
+            synthetic_data (dict[str, pandas.DataFrame]):
+                The tables from the synthetic dataset, passed as a dictionary of
+                table names and pandas.DataFrames.
+            metadata (dict):
+                Multi-table metadata dict. If not passed, it is build based on the
+                real_data fields and dtypes.
+            statistic (str):
+                The desired statistic to compute. Must be either 'mean', 'median', or 'std'.
+
+        Returns:
+            float:
+                The average of all (parent, child) cardinality statistic similarity scores.
+        """
+        score_breakdowns = cls.compute_breakdown(real_data, synthetic_data, metadata, statistic)
+        if 'score' in score_breakdowns:
```

Comment:
What is the situation where score ends up on the outside dict?

Reply:
It's if there are no relationships in the metadata (the metric is not applicable). In that case we set the metric score to `np.nan`.